### PR TITLE
Support Custom Typed Array

### PR DESCRIPTION
**Pull request by joshuajiangdev**
Context: https://github.com/argentlabs/web3.swift/issues/277

Adding support of custom typed array in `signableHash`. Cross verified new unit test cases with [@metamask/eth-sig-util](https://github.com/MetaMask/eth-sig-util).

**Tests:**
1. All unit tests in `EthereumAccount_SignTypedTests`

### CODE REVIEW

**Comment by joshuajiangdev** [comment]
Hmmm, seems the `unexpected` failures are not from the unit tests I modified 
```
Test Suite 'EthereumWebSocketClientTests' failed at 2022-11-16 15:14:59.359.
	 Executed 33 tests, with 35 failures (3 unexpected) in 5.265 (5.267) seconds
```
Any suggestion here?

File: web3swift/src/Account/TypedData.swift [hunk 1]
```diff
@@ -71,7 +71,11 @@ extension TypedData {
             let param = types[type]!.map { "\($0.type) \($0.name)" }.joined(separator: ",")
             return "\(type)(\(param))"
         }.joined()
-        return encoded.data(using: .utf8) ?? Data()
+        
+        let result = encoded.data(using: .utf8) ?? Data()
+        
+        print("\(primaryType) type encode: \(result.web3.hexString)")
```
Comment (dnKaratzas) [inline web3swift/src/Account/TypedData.swift:77]: Please remove.

**Review by dnKaratzas — APPROVED** [review]
LGTM!

File: web3swift/src/Account/TypedData.swift (hunk 1 above)
Comment (joshuajiangdev) [reply web3swift/src/Account/TypedData.swift]: Done.

**Comment by joshuajiangdev** [comment]
Hmmmm `1 workflow awaiting approval` has been stucked for 1 day. How to resolve this?

**Review by DarthMike — APPROVED** [review]

**Comment by joshuajiangdev** [comment]
in macos workflow, I'm seeing
```
Test Suite 'EthereumWebSocketClientTests' failed at 2022-11-18 08:45:31.549.
	 Executed 33 tests, with 35 failures (3 unexpected) in 4.931 (4.933) seconds
```

in linux workflow, I'm seeing
```
Test Suite 'All tests' failed at 2022-11-18 09:10:51.236
	 Executed 307 tests, with 111 failures (0 unexpected) in 1634.268 (1634.268) seconds
Error: Process completed with exit code 1.
```

Two questions here:
1. Linux workflow seems do not have unexpected errors, but it still shows blocked/failed?
2. The macos workflow seems also have unexpected fails without my change eg. develop branch.

If both failures are expected, how could I land with two workflows to be in the failure state?

**Comment by DarthMike** [comment]
@joshuajiangdev We need to migrate to Goerli for tests to pass. Will take this task this week and we can update your PR with that changes so tests pass. Will let you know about progress (https://github.com/argentlabs/web3.swift/issues/234)

**Comment by DarthMike** [comment]
@joshuajiangdev #281 should fix the tests

**Pull request merged**